sched/sched/sched_timerexpiration.c

```diff
@@ -388,6 +391,8 @@ static clock_t nxsched_timer_start(clock_t ticks, clock_t interval)
         }
 #endif
 
+      flags = enter_critical_section();
```

Comment:
why mix the critical section and spinlock?

Reply:
Perhaps we need to protect up_timer_tick_start.